{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"YOLO_Validate_full_yolo_liveNuclei.ipynb","provenance":[{"file_id":"1M3vBxvkuVe9gAnJiYFRo8k03eIhCyELj","timestamp":1603816934017}],"collapsed_sections":[],"authorship_tag":"ABX9TyN+L2Z7L04zaUaw5FDZ4ve4"},"kernelspec":{"name":"python3","display_name":"Python 3"},"accelerator":"GPU"},"cells":[{"cell_type":"code","metadata":{"id":"3X8NSero2PYH","executionInfo":{"status":"ok","timestamp":1603816275943,"user_tz":240,"elapsed":29671,"user":{"displayName":"Anthony Giampetruzzi","photoUrl":"","userId":"08675423877236153792"}},"outputId":"8e224a9c-e8e9-49ca-de9b-50152dbe621d","colab":{"base_uri":"https://localhost:8080/","height":323}},"source":["!pip install imgaug\n","from google.colab import drive\n","drive.mount('/content/gdrive')\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Requirement already satisfied: imgaug in /usr/local/lib/python3.6/dist-packages (0.2.9)\n","Requirement already satisfied: matplotlib in /usr/local/lib/python3.6/dist-packages (from imgaug) (3.2.2)\n","Requirement already satisfied: opencv-python in /usr/local/lib/python3.6/dist-packages (from imgaug) (4.1.2.30)\n","Requirement already satisfied: six in /usr/local/lib/python3.6/dist-packages (from imgaug) (1.15.0)\n","Requirement already satisfied: Shapely in /usr/local/lib/python3.6/dist-packages (from imgaug) (1.7.1)\n","Requirement already satisfied: numpy>=1.15.0 in /usr/local/lib/python3.6/dist-packages (from imgaug) (1.18.5)\n","Requirement already satisfied: Pillow in /usr/local/lib/python3.6/dist-packages (from imgaug) (7.0.0)\n","Requirement already satisfied: imageio in /usr/local/lib/python3.6/dist-packages (from imgaug) (2.4.1)\n","Requirement already satisfied: scikit-image>=0.11.0 in /usr/local/lib/python3.6/dist-packages (from imgaug) (0.16.2)\n","Requirement already satisfied: scipy in /usr/local/lib/python3.6/dist-packages (from imgaug) (1.4.1)\n","Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/local/lib/python3.6/dist-packages (from matplotlib->imgaug) (2.4.7)\n","Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.6/dist-packages (from matplotlib->imgaug) (0.10.0)\n","Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.6/dist-packages (from matplotlib->imgaug) (1.2.0)\n","Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.6/dist-packages (from matplotlib->imgaug) (2.8.1)\n","Requirement already satisfied: PyWavelets>=0.4.0 in /usr/local/lib/python3.6/dist-packages (from scikit-image>=0.11.0->imgaug) (1.1.1)\n","Requirement already satisfied: networkx>=2.0 in /usr/local/lib/python3.6/dist-packages (from scikit-image>=0.11.0->imgaug) (2.5)\n","Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.6/dist-packages (from networkx>=2.0->scikit-image>=0.11.0->imgaug) (4.4.2)\n","Mounted at /content/gdrive\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"Fgkchpod29bl","executionInfo":{"status":"ok","timestamp":1603816308064,"user_tz":240,"elapsed":29139,"user":{"displayName":"Anthony Giampetruzzi","photoUrl":"","userId":"08675423877236153792"}},"outputId":"e40fa3bc-2a41-4b8b-b985-05a5bf03fed3","colab":{"base_uri":"https://localhost:8080/","height":1000}},"source":["!cp -r '/content/gdrive/My Drive/YOLO_TrainingTools/.'  /content/.\n","!mkdir /content/Validation_Set/\n","!cp -r '/content/gdrive/My Drive/Validation_Set/.'  /content/Validation_Set/\n","!python yolo_validate.py -c config.json -w full_yolo_liveNuclei.h5 -i /content/Validation_Set/\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["2020-10-27 16:31:39.306225: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1\n","2020-10-27 16:31:40.668922: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1\n","2020-10-27 16:31:40.719316: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:40.719917: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n","coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n","2020-10-27 16:31:40.719974: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1\n","2020-10-27 16:31:40.946406: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.10\n","2020-10-27 16:31:41.062042: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10\n","2020-10-27 16:31:41.108844: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10\n","2020-10-27 16:31:41.400345: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10\n","2020-10-27 16:31:41.425134: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10\n","2020-10-27 16:31:41.940015: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7\n","2020-10-27 16:31:41.940236: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:41.940899: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:41.941438: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0\n","2020-10-27 16:31:42.002077: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 2200000000 Hz\n","2020-10-27 16:31:42.002299: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x2d1ebc0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n","2020-10-27 16:31:42.002327: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n","2020-10-27 16:31:42.141756: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:42.142479: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x2d1ed80 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n","2020-10-27 16:31:42.142511: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5\n","2020-10-27 16:31:42.143135: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:42.143695: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla T4 computeCapability: 7.5\n","coreClock: 1.59GHz coreCount: 40 deviceMemorySize: 14.73GiB deviceMemoryBandwidth: 298.08GiB/s\n","2020-10-27 16:31:42.143769: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1\n","2020-10-27 16:31:42.143824: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcublas.so.10\n","2020-10-27 16:31:42.143847: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10\n","2020-10-27 16:31:42.143874: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10\n","2020-10-27 16:31:42.143903: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10\n","2020-10-27 16:31:42.143924: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10\n","2020-10-27 16:31:42.143946: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7\n","2020-10-27 16:31:42.144025: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:42.144608: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:42.145103: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0\n","2020-10-27 16:31:42.147477: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1\n","2020-10-27 16:31:45.932244: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1257] Device interconnect StreamExecutor with strength 1 edge matrix:\n","2020-10-27 16:31:45.932300: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1263]      0 \n","2020-10-27 16:31:45.932313: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1276] 0:   N \n","2020-10-27 16:31:45.935212: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:45.935879: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-10-27 16:31:45.936460: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n","2020-10-27 16:31:45.936503: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1402] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 13962 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n","Model: \"functional_3\"\n","__________________________________________________________________________________________________\n","Layer (type)                    Output Shape         Param #     Connected to                     \n","==================================================================================================\n","input_1 (InputLayer)            [(None, 416, 416, 3) 0                                            \n","__________________________________________________________________________________________________\n","functional_1 (Functional)       (None, 13, 13, 1024) 50547936    input_1[0][0]                    \n","__________________________________________________________________________________________________\n","conv_23 (Conv2D)                (None, 13, 13, 30)   30750       functional_1[0][0]               \n","__________________________________________________________________________________________________\n","reshape (Reshape)               (None, 13, 13, 5, 6) 0           conv_23[0][0]                    \n","__________________________________________________________________________________________________\n","input_2 (InputLayer)            [(None, 1, 1, 1, 10, 0                                            \n","__________________________________________________________________________________________________\n","lambda_1 (Lambda)               (None, 13, 13, 5, 6) 0           reshape[0][0]                    \n","                                                                 input_2[0][0]                    \n","==================================================================================================\n","Total params: 50,578,686\n","Trainable params: 50,558,014\n","Non-trainable params: 20,672\n","__________________________________________________________________________________________________\n","full_yolo_fl.h5\n","Traceback (most recent call last):\n","  File \"yolo_validate.py\", line 231, in <module>\n","    _main_(args)\n","  File \"yolo_validate.py\", line 195, in _main_\n","    yolo.load_weights(weights_path)\n","  File \"/content/frontend.py\", line 245, in load_weights\n","    self.model.load_weights(weight_path)\n","  File \"/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py\", line 2204, in load_weights\n","    with h5py.File(filepath, 'r') as f:\n","  File \"/usr/local/lib/python3.6/dist-packages/h5py/_hl/files.py\", line 408, in __init__\n","    swmr=swmr)\n","  File \"/usr/local/lib/python3.6/dist-packages/h5py/_hl/files.py\", line 173, in make_fid\n","    fid = h5f.open(name, flags, fapl=fapl)\n","  File \"h5py/_objects.pyx\", line 54, in h5py._objects.with_phil.wrapper\n","  File \"h5py/_objects.pyx\", line 55, in h5py._objects.with_phil.wrapper\n","  File \"h5py/h5f.pyx\", line 88, in h5py.h5f.open\n","OSError: Unable to open file (unable to open file: name = 'full_yolo_fl.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)\n"],"name":"stdout"}]}]}